[Seethevon] iP

Seethevon opened the pull request:
# BibiPro 🐱 
BibiPro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn

>All you need to do is:
1. download it from [here](https://nus-cs2103-ay2425s2.github.io/website/admin/ip-w4.html)
2. double-click it
3. start adding your tasks

and its all **FREE**!

Features:
- [x] Manages tasks
- [x] Manages deadlines

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
``` Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
level-0

Commit:
test

Commit:
Revert "test"

This reverts commit 7834638334f95928418c2794e3cc9b08c3b99958.

Commit:
level0

Commit:
level-1

Commit:
level-2

Commit:
level-3

Commit:
level-4

Commit:
A-TextUiTesting

Commit:
level-5

Commit:
adding the delete function to the bot

Commit:
implementing save level-7

Commit:
implementing date & time level-8

Commit:
Making it more oop

Commit:
Package

Commit:
no message

Commit:
Merge branch 'add-gradle-support'

Commit:
adding test casses

Commit:
correcting the test casses

Commit:
no message

Commit:
creating JAR file

Commit:
fixed some outputs

Commit:
Add header comments to non-private classes/methods

Commit:
Add the find function

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Bibi/Parser.java

kerktaiheng commented on src/main/java/Bibi/TaskList.java:
```diff
@@ -56,4 +56,27 @@ public Task getTask(int index) {
+
+        return sb.toString();
+    }
 }
```
Don't forget to end with newline

kerktaiheng commented on src/main/java/Bibi/Ui.java:
```diff
@@ -27,4 +27,10 @@ public void showMessage(String message) {
+        System.out.println(result);
+        System.out.println("____________________________________________________________");
+    }
 }
```
Same as above
Additionally, the divider can be a static final field so that you are guaranteed consistency and have cleaner code

kerktaiheng commented on src/main/java/Bibi/Task.java:
```diff
@@ -29,6 +29,6 @@ public String toString() {
-        return "Bibi.Bibi.Task | " + (isDone ? "1" : "0") + " | " + description;
+        return "Task | " + (isDone ? "1" : "0") + " | " + description;
     }
 }
```
Missing newline

kerktaiheng commented on src/test/java/Bibi/DeadlineTest.java:
```diff
@@ -4,6 +4,13 @@
 import java.time.LocalDateTime;
 
 public class DeadlineTest {
+    @Test
```
In general may wanna add some docs to the test so we are clearer with the intent
not the point of the PR anyway but randomly generated data would be good

kerktaiheng reviewed (commented):
In general the style is great! High effort stuff

Most of the issues with style can be automated away anyway.

Very cute bot meow

datletu reviewed (commented):
Overall I didn't find any issue with the naming of variables. Can you separate the different classes into different packages for the code to be more organized? Besides that nice job lol

Commit:
Implement a GUI

Commit:
Remove redundant code

Commit:
Create text file for storage if file does not exist

Commit:
Add asserts

Commit:
Refactor classes and their methods

Seethevon committed:
Merge pull request #2 from Seethevon/branch-A-Assertions

Add asserts

Seethevon committed:
Merge pull request #3 from Seethevon/branch-A-CodeQuality

Refactor classes and their methods

Commit:
Add snooze and help extension

Commit:
BCD-Extenstion: Add snooze command to UI

The UI does not have a way for users to postpone tasks. Users may need to reschedule deadlines.

Let's add a snooze command that allows users to defer deadlines to a later time. This improves flexibility in task management.

Commit:
Add Javadoc comments to multiple classes

Several classes across the project lacked proper documentation.

Adding Javadoc comments enhances code readability and provides a clear understanding of the methods, parameters, and return types.

Let's:
- Add descriptions for key methods and classes
-Document method parameters, return types, and exceptions
-Improve the maintainability and clarity of the codebase

Seethevon committed:
Update README.md

Seethevon committed:
Add files via upload

Seethevon committed:
Update README.md

Seethevon committed:
Update README.md

Seethevon committed:
Update README.md

Seethevon committed:
Update README.md

Seethevon committed:
Add files via upload

Seethevon committed:
Add files via upload

Seethevon committed:
Update README.md

Seethevon committed:
Add files via upload

Seethevon committed:
Update README.md

Seethevon committed:
Update README.md

Seethevon committed:
Rename Ui.PNG to Ui.png
